lib/tcp/tcp_packet.ml

```diff
@@ -73,7 +73,6 @@ module Marshal = struct
   let unsafe_fill ~pseudoheader ~payload t buf options_len =
     let data_off = (sizeof_tcp / 4) + (options_len / 4) in
     let buf = Cstruct.sub buf 0 (data_off * 4) in
-    Cstruct.memset buf 0; (* autopad *)
```

Comment:
The `(* autopad *)` comment is a bit cryptic -- what I meant here was "we may have to include some padding in the TCP options, so make sure this padding is set to `\000` as is appropriate."  @avsm is right that this patch was merged shortly before `Cstruct.create` was changed to zero memory by default; `tcpip` (and indeed many `Cstruct` users) could use another look to see whether there are unnecessary `memset` calls generally.